Issue #34 - space encoded as plus

homes2001 opened the pull request.

thesmith commented on src/main/java/vc/inreach/aws/request/AWSSigningRequestInterceptor.java:
```diff
@@ -36,7 +39,13 @@ public void process(HttpRequest request, HttpContext context) throws HttpExcepti
     private Multimap<String, String> params(HttpRequest request) {
-        return params(((HttpRequestWrapper) request).getURI().getQuery());
+        try {
+            final String rawQuery = ((HttpRequestWrapper) request).getURI().getRawQuery();
```
This will often be null (and throw an NPE on the next line)

thesmith commented:
Can this be squashed down or something? Not sure why there's the first commit...

homes2001 commented:
You are right, there is a NPE when there is no query string. I have added a test case to verify and to make sure this does not happen in the future anymore. Furthermore I have removed the first commit. Also I have removed the try/catch to make the code cleaner, as the calling function can cope with IOExceptions.

homes2001 commented:
@thesmith did you have a chance to review the new commits?

thesmith commented:
All done!
